modules/geo-layers/src/mesh-layer/mesh-layer.js

```diff
@@ -124,6 +130,18 @@ export default class _MeshLayer extends SimpleMeshLayer {
 
     attribute.value = value;
   }
+
+  destroyMaterialParserTextures() {
+    if (this.state.materialParser) {
+      this.state.materialParser.destroyTextures();
+      this.setState({materialParser: null});
+    }
```

Comment:
Why not just call `destroy()` / `delete()` and create a new materialParser? I am not sure we even need this helper method.

```suggestion
    this.state.materialParser?.destroy();
    this.setState({materialParser: null});
```

Reply:
done